.pipelines/azdo-abtest-pipeline.yml

```diff
@@ -0,0 +1,162 @@
+      artifact: allcharts
+
+
+- stage: 'Blue_Staging'
+  jobs:
+  - deployment: "Deploy_to_Staging"
+    timeoutInMinutes: 0
+    environment: abtestenv
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - script: |
+              IMAGE_LOCATION="$(cat $(Pipeline.Workspace)/image_location/image_location.txt)"
+              echo "##vso[task.setvariable variable=IMAGE_LOCATION]$IMAGE_LOCATION"
+            displayName: 'Get Image Location'
+          - template: azdo-helm-upgrade.yml
+            parameters:
+              chartPath: '$(Pipeline.Workspace)/allcharts/abtest-model'
+              releaseName: $(blueReleaseName) 
+              overrideValues: 'deployment.name=$(blueReleaseName),deployment.bluegreen=blue,deployment.image.name=$(IMAGE_LOCATION)'
+
+- stage: 'Blue_50'
+  jobs:
+  - job: 'Blue_Rollout_50'
+    displayName: 50 50 rollout to blue environment
+    timeoutInMinutes: 0
+    steps:
+    - template: azdo-helm-upgrade.yml
+      parameters:
+        chartPath: '$(System.DefaultWorkingDirectory)/charts/abtest-istio'
+        releaseName: 'abtest-istio' 
+        overrideValues: 'weight.blue=50,weight.green=50'
+
+- stage: 'Blue_100'
+  jobs:
+  - deployment: 'blue_Rollout_100'
+    timeoutInMinutes: 0
+    environment: abtestenv
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - template: azdo-helm-upgrade.yml
+            parameters:
+              chartPath: '$(Pipeline.Workspace)/allcharts/abtest-istio'
+              releaseName: 'abtest-istio' 
+              overrideValues: 'weight.blue=100,weight.green=0'
+
+- stage: 'Rollback'
+  dependsOn: 'Blue_100'
+  condition: failed()
+  jobs:
+  - deployment: 'Roll_Back'
+    displayName: 'Roll Back after failure'
+    environment: abtestenv
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - template: azdo-helm-upgrade.yml
+            parameters:
+              chartPath: '$(Pipeline.Workspace)/allcharts/abtest-istio'
+              releaseName: 'abtest-istio' 
+              overrideValues: 'weight.blue=0,weight.green=100'
+
+- stage: 'Set_Production_Tag'
+  dependsOn: 'Blue_100'
+  condition: succeeded()
+  jobs:
+  - deployment: 'green_blue_tagging'
+    timeoutInMinutes: 0
+    environment: abtestenv
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - script: |
+              IMAGE_LOCATION="$(cat $(Pipeline.Workspace)/image_location/image_location.txt)"
+              echo "##vso[task.setvariable variable=IMAGE_LOCATION]$IMAGE_LOCATION"
+            displayName: 'Get Image Location'
+          - template: azdo-helm-upgrade.yml
+            parameters:
+              chartPath: '$(Pipeline.Workspace)/allcharts/abtest-model'
+              releaseName: $(greenReleaseName) 
+              overrideValues: 'deployment.name=$(greenReleaseName),deployment.bluegreen=green,initialDeployment=true,deployment.image.name=$(IMAGE_LOCATION)'
```

Comment:
`intialDeployment=true` ok every time?

Reply:
Good catch. it's garbage. Not used anymore